napari/_qt/qt_main_window.py

```diff
@@ -378,6 +378,36 @@ def __init__(self, viewer, *, show: bool = True):
                 self.qt_viewer.canvas._backend.screen_changed
             )
 
+        # set the grid options on start up
+
+        stride_ge = settings.application.__fields__['grid_stride'].type_.ge
+        stride_le = settings.application.__fields__['grid_stride'].type_.le
+        stride_ne = settings.application.__fields__['grid_stride'].type_.ne
+
+        self.qt_viewer.viewer.grid.__fields__['stride'].type_.ge = stride_ge
```

Comment:
hmm... _assigning_ to `type_.ge` or `type_.le` is not quite right here: you're effectively just changing/undoing what you've declared [on the grid model](https://github.com/napari/napari/pull/3084/files#diff-659b3b20559ec48ce2510d61878848e4e63be2f77988857dd404cc3f24f899cbR33-R34) here...  
don't think of these as "current values"... think of this as constraints that are set once (on model declaration) and never touched again.  The widget then can have a current value that changes within these constraints... 

can you tell me what you're going for here?  Are you trying to initialize a widget?

Reply:
@tlambert03 I wanted to be able to set these values in 1 place and not have to change them somewhere else.  If someone wanted to adjust min and max of stride, for example, right now we want to do it in 2 places?  I realize we won't be doing this many times, but if someone did want to down the line, then we need to make note that they should make the changes in settinggs _defaults.py and components grid.py.

Reply:
I will take this stuff out for now.  I just feel uncomfortable having the same information in 2 separate places, unless you have a suggestion on how to avoid that?

Reply:
> right now we want to do it in 2 places

if you're trying to avoid having both 
```py
class ApplicationSettings:
    grid_stride: conint(ge=-50, le=50, ne=0) = ...

class GridCanvas:
    stride: conint(ge=-50, le=50, ne=0) = ...
```

then you can use a type variable:
```py

GridStride = conint(ge=-50, le=50, ne=0)

class ApplicationSettings:
    grid_stride: GridStride = ...

class GridCanvas:
    stride: GridStride = ...
```

Reply:
another possibility you could try here is to make `GridCanvas.stride` an `@property` that just gets the current Application setting?

Reply:
I went with the option A.  I don't want to do a risky thing right now.  😆